app/controllers/interactive_pages_controller.rb

```diff
@@ -13,18 +13,16 @@ class InteractivePagesController < ApplicationController
   def show
     authorize! :read, @page
     if !params[:run_key]
-      redirect_to page_with_run_path(@activity.id, @page.id, @run_key, request.query_parameters) and return
-    end
-
-    begin
-      authorize! :access, @run
-    rescue
-      user_id_mismatch()
-      render 'runs/unauthorized_run'
-      return
+      if @sequence
+        redirect_to sequence_page_with_run_path(@sequence, @activity.id, @page.id, @run_key, request.query_parameters) and return
+      else
+        redirect_to page_with_run_path(@activity.id, @page.id, @run_key, request.query_parameters) and return
+      end
```

Comment:
Since this logic of checking for the existence of `@sequence` is used in several places. It seems like this should be moved into a helper. So the helper is something like:  `page_with_run_path_helper(@sequence, @activity.id, page.id, @run_key, request.query_parameters)` and its code checks if @sequence is set or not.